packages/react-utils/src/useControlled.ts

```diff
@@ -0,0 +1,106 @@
+// @see https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/useControlled
+/* eslint-disable react-hooks/rules-of-hooks, react-hooks/exhaustive-deps -- Based on Mui */
+/* eslint-disable tsdoc/syntax -- Allow documentation of object params */
+
+'use client'
```

Comment:
I wonder what happens during buildtime with the `'use client'` directive, since it's not at the top-top of the file? 🤔

Reply:
Nice one, just verified that the `'use client'` directive is preserved in the `dist` folder 👍